firebase-firestore/src/main/java/com/google/firebase/firestore/core/Target.java

```diff
@@ -211,13 +208,8 @@ public Bound getLowerBound(FieldIndex fieldIndex) {
             filterInclusive = false;
             break;
           case NOT_EQUAL:
-            filterValue = Values.MIN_VALUE;
-            break;
           case NOT_IN:
-            filterValue =
-                Value.newBuilder()
-                    .setArrayValue(ArrayValue.newBuilder().addValues(MIN_VALUE))
-                    .build();
+            filterValue = Values.MIN_VALUE;
```

Comment:
Not In and Not Equal use the same bound, which makes the comparison logic at the end of this method work as we do not want to compare to an array but to an actual minimum.